lib/transports/rest2.js

```diff
@@ -250,6 +254,36 @@ class RESTv2 {
     return new ModelClass(data)
   }
 
+  /**
+   * @param {Arr} data order matters
+   * @return {Arr} merged arr of coins and names with no pairs rpeated
+   * @private
+   */
+  _mergeAllArrOfCoins (data) {
+    const addKeyToObj = (obj, curr) => {
+      if (!Array.isArray(curr)) obj[curr] = curr
+      else if (curr.length === 1) obj[curr[0]] = curr[0]
+      else obj[curr[0]] = curr[1]
+
+      return obj
+    }
+
+    const arrOfKeyObjects = _transform(data, (allObjects, arr, ii) => {
```

Comment:
Can you simplify this, and loop over the objects with a traditional `for`, maybe using a temporary object to ensure no duplicates? It's difficult to follow what is happening